### PR TITLE
Support glyph source status color code in UFO/Designspace

### LINKED ISSUE #1364: [designspace backend] Support reading/writing the development status code to/from UFO/designspace
This may involve implementing a general way to store GlyphSource.customData as a whole in the source .glif. (Layer instances also have a customData attribute, which is distinct. Would be nice to solve both, though.)

Also, check whether we read the designspace lib correctly, which is where I expect the dev status definitions to be stored.

Relates to #1360.

### PR DESCRIPTION
**Pull request by ollimeier**
Fixes #1364

### CODE REVIEW

**Review by justvanrossum — COMMENTED** [review]
I'm afraid this isn't quite going in the right direction:
- The source's customData should be stored in the glyph lib of the _associated_ ufo glyph, not collected in the default glyph
- Any tests need to go to test_backends_designspace.py, not test_backends_fontra.py, which is for testing the .fontra backend

File: test-py/test_backends_fontra.py [hunk 1]
```diff
@@ -136,3 +136,23 @@ async def test_statusFieldDefinitions(writableFontraFont):
     await writableFontraFont.putCustomData(statusTestData)
 
     assert statusTestData == await writableFontraFont.getCustomData()
+
+
+async def test_getGlyphSourceStatusCode(testFontraFont):
```
Comment (justvanrossum) [inline test-py/test_backends_fontra.py:141]: These tests are redundant, as there is no special support needed for customData or status codes. Please remove these tests.

File: test-common/fonts/MutatorSans.fontra/glyphs/E^1.json [hunk 2]
```diff
@@ -8,6 +8,9 @@
 "italic": 0,
 "weight": 150,
 "width": 0
+},
```
Comment (justvanrossum) [inline test-common/fonts/MutatorSans.fontra/glyphs/E^1.json:11]: Together with the redundant tests in tests_backends_fontra.py, this change can be reverted, too.

File: src/fontra/backends/designspace.py [hunk 3]
```diff
@@ -459,14 +474,22 @@ async def putGlyph(
...
 
+            customData = None
             if glyphSet == self.defaultUFOLayer.glyphSet:
                 layerGlyph = defaultLayerGlyph
                 storeInLib(layerGlyph, GLYPH_DESIGNSPACE_LIB_KEY, localDS)
                 storeInLib(layerGlyph, SOURCE_NAME_MAPPING_LIB_KEY, sourceNameMapping)
                 storeInLib(layerGlyph, LAYER_NAME_MAPPING_LIB_KEY, layerNameMapping)
-                storeInLib(layerGlyph, CUSTOM_DATA_LIB_KEY, glyph.customData)
+                customData = (
+                    glyph.customData | localSourcesCustomData[ufoLayer.fontraLayerName]
```
Comment (justvanrossum) [inline src/fontra/backends/designspace.py:484]: We need to differentiate between these two, we can't just fold them into one. 

Suggest to use two keys:
```python
GLYPH_CUSTOM_DATA_LIB_KEY = "xyz.fontra.glyph.customData"
GLYPH_SOURCE_CUSTOM_DATA_LIB_KEY = "xyz.fontra.glyph.source.customData"
```

File: src/fontra/backends/designspace.py [hunk 4]
```diff
@@ -459,14 +474,22 @@ async def putGlyph(
...
+                )
             else:
                 layerGlyph = readGlyphOrCreate(glyphSet, glyphName, codePoints)
+                if localSourcesCustomData.get(ufoLayer.fontraLayerName) is not None:
```
Comment (justvanrossum) [inline src/fontra/backends/designspace.py:488]: General Python advice:
- To check if a key exists in a dict, you write "if key in dict"
- To get a key with a fallback, use "dict.get(key)"

You _either_ first check if a key exists, and then retrieve it normally, or you "get" it and then check the value for None. You are using a mix of both here.

File: test-py/test_backends_fontra.py (hunk 1 above)
Comment (ollimeier) [reply test-py/test_backends_fontra.py]: done.

File: test-common/fonts/MutatorSans.fontra/glyphs/E^1.json (hunk 2 above)
Comment (ollimeier) [reply test-common/fonts/MutatorSans.fontra/glyphs/E^1.json]: done.

File: src/fontra/backends/designspace.py (hunk 3 above)
Comment (ollimeier) [reply src/fontra/backends/designspace.py]: Thank you very much for the valid feedback. I changed it.

File: src/fontra/backends/designspace.py (hunk 4 above)
Comment (justvanrossum) [reply src/fontra/backends/designspace.py]: Actually, you just moved it in [this commit](https://github.com/googlefonts/fontra/pull/1403/commits/34e934aa51a03837d783ee84d5ae00100cb9d086)... I fixed it (together with a general logic error) [here](https://github.com/googlefonts/fontra/pull/1403/commits/7ff723a36ed16c4a22dd417022e89dc713801ef5).

**Pull request merged**